sec.club/src/articles/events/pimp-my-terminal/pimp-my-terminal.md

```diff
@@ -0,0 +1,370 @@
+
+### Are there any packages I should install to make things easier? <small style="opacity: 0.5">(Unix only)</small>
+
+It would be useful to have either `curl` or `wget` installed.
+On Mac, `curl` should be installed by default. These programs
+let you download files from the internet. Useful for downloading
+configs directly into the right spot.
+
+# Time Savers
+
+### Downloading config files
+
+You may want to copy somebody else's config from the internet.
+However, selecting all the text, copying it, finding the file
+it goes in, pasting it, etc, can be a bit of a pain after you
+do it multiple times.
+
+#### Windows
+
+```powershell
+Invoke-WebRequest -Uri URL -OutFile TARGET
+```
+
+#### Unix
+
+Using wget:
+
+```bash
+wget -O TARGET URL
+```
+
+Using curl:
+```bash
+curl -o TARGET URL
+```
+
+# IDEs
+
+Your IDE is the easiest place to start making your environment
+something that works for you.
+
+### VS Code
+
+[Download](https://code.visualstudio.com) VS Code here.
+
+[How to set your theme in VS Code](/tutorial/vscode-theme)
+
+#### Extensions
+
+*Tip: You can install these quickly by pressing `Ctrl + P` and typing the command `ext install PACKAGE`*
+
+- TabNine `tabnine.tabnine-vscode` - Uses machine learning to help you write code faster. - Not battery friendy, not memory friendly.
+- GitLens `eamodio.gitlens` - Extra git integration features.
+- Path Intellisense `christian-kohler.path-intellisense` - Autocompletes filenames.
+- Debug Visualizer `hediet.debug-visualizer` - Fancy graphics for visualizing data structures while stepping through code.
+- Vim for VS Code `vscodevim.vim` - Vim keybinds for VS Code.
+
+### Atom
+
+Similar to VS Code. Very customizable.
+
+[Download](https://atom.io) Atom here.
+
+#### Extensions
+
+- [Hydrogen](https://atom.io/packages/hydrogen) - Turn your python files into jupyter notebooks.
+
+### Sublime Text
+
+In the same vein as VS Code and Atom.
+
+[Download](https://www.sublimetext.com) Subtime Text here.
+
+### Vim
+
+Vim is a text editor that runs in your terminal. [Check out how to use Vim here](/tutorial/cli-text-editors)
+
+Vim configuration is found at `~/.vimrc`
+
+- https://github.com/amix/vimrc - Useful preset configuration for Vim
+- [Check out Vim color schemes here](https://github.com/rafi/awesome-vim-colorschemes)
+
+### Emacs
+
+Emacs is similar to Vim, but it doesn't run in your terminal.
+
+[Learn how to customize Emacs here](https://www.gnu.org/software/emacs/manual/html_node/emacs/Easy-Customization.html)
+
+# Terminals
+
+You need to have a terminal in order to pimp it out! You
+probably already have one, but if it's the default one
+(especially on windows) it probably sucks.
+
+### Windows
+
+Windows users should take a look at the [new Windows terminal from Microsoft](https://github.com/microsoft/terminal).
+You can install it from the Windows Store.
+
+There are other options:
+
+- [Windows Terminal](https://github.com/microsoft/terminal)
```

Comment:
I think its a little confusing to list Windows Terminal as another option when it was just discussed above

Reply:
My thinking was that if somebody was just glossing over the content and not actually reading it, they wouldn't see windows terminal listed above

Reply:
Fair enough